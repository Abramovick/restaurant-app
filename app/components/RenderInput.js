import React from 'react';
import PropTypes from 'prop-types';
import { Input, Item, Icon } from 'native-base';
import { HomeScreenStyles as styles } from '../styles/styles';

const RenderInput = ({
  placeholder,
  meta: { touched, error },
  input: { onChange, ...restInput }
}) =>
    <Item style={(touched && error) ?
      [styles.inputView, styles.errorStyle] : styles.inputView}>
      {(touched && error) && <Icon style={styles.searchIcon} name='alert'/>}
      <Input
      style={styles.input}
      onChangeText={onChange}
      {...restInput}
      placeholder={placeholder}
      placeholderTextColor="white"/>
    </Item>;

RenderInput.propTypes = {
  placeholder: PropTypes.string,
};

RenderInput.propTypes = {
  meta: PropTypes.object,
  input: PropTypes.object,
  placeholder: PropTypes.string,
  onChangeText: PropTypes.func,
};

export default RenderInput;
