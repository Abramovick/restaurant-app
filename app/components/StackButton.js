import React from 'react';
import PropTypes from 'prop-types';
import { TouchableHighlight } from 'react-native';
import { Icon } from 'native-base';
import { StackButtonStyles as styles } from '../styles/styles';

const StackButton = navigation =>
<TouchableHighlight
  underlayColor='#D57E56'
  onPress={() => {
      navigation.navigate('DrawerStack');
  }
  }>
  <Icon
  style={styles.icon}
  name='ios-arrow-round-back-outline'/>
</TouchableHighlight>;

StackButton.propTypes = {
  navigation: PropTypes.object,
};

export default StackButton;
