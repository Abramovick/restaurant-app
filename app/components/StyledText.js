import React from 'react';
import { Text } from 'react-native';
import PropTypes from 'prop-types';

export const StyledText = props =>
  <Text
  {...props}
    style={[props.style, { fontFamily: 'SinkinSans-200XLight' }]}
  />;

StyledText.propTypes = {
  style: PropTypes.object
};
