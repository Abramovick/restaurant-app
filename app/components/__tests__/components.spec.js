import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import RenderInput from '../RenderInput';
import StackButton from '../StackButton';
import { StyledText } from '../StyledText';

const meta = {
  touched: true,
  error: true
};
const metaFalse = {
  touched: false,
  error: false
};

const input = {
  onChange: jest.fn()
};

const navigation = {};
const navigate = jest.fn();
navigation.navigate = me => me;
describe('RENDERINPUT --- Snapshot', () => {
  it('renders the cart screen if user is logged in', async () => {
    const tree = renderer.create(<RenderInput
      placeholder={'items'}
      meta={meta}
      input={input}
      secureTextEntry
      />);
    expect(tree).toMatchSnapshot();
  });
  it('renders the render input', async () => {
    const tree = renderer.create(<RenderInput
      placeholder={'items'}
      meta={metaFalse}
      input={input}
      secureTextEntry
      />);
    expect(tree).toMatchSnapshot();
  });
});

describe('STACKBUTTON --- Snapshot', () => {
  it('renders the stack button', async () => {
    const container = shallow(<StackButton navigate={navigate}/>);
    container.find('TouchableHighlight').first().simulate('press');
  });
});

describe('STYLEDTEXT --- Snapshot', () => {
  it('renders the styled text', async () => {
    const tree = renderer.create(<StyledText
      color={'red'}
      customFont={'Sans'}
      smallFont/>);
    expect(tree).toMatchSnapshot();
  });

  it('renders the styled text', async () => {
    const tree = renderer.create(<StyledText
      color={'red'}
      customFont={'Sans'}
      />);
    expect(tree).toMatchSnapshot();
  });
});

describe('VALIDATEINPUT --- Snapshot', () => {
  it('renders the styled text', async () => {
    const tree = renderer.create(<StyledText
      color={'red'}
      customFont={'Sans'}
      smallFont/>);
    expect(tree).toMatchSnapshot();
  });
});
