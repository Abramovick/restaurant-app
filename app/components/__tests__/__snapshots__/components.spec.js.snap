// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RENDERINPUT --- Snapshot renders the input with error if touched and error are false 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E0956D",
      "borderBottomWidth": 1,
      "borderColor": "#D9D5DC",
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopWidth": 0,
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "center",
      "marginLeft": 2,
      "marginTop": "3%",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <TextInput
    allowFontScaling={true}
    editable={true}
    onChangeText={[Function]}
    placeholder="items"
    placeholderTextColor="white"
    secureTextEntry={true}
    style={
      Array [
        Object {
          "color": "#000",
          "flex": 1,
          "fontSize": 17,
          "height": 50,
          "lineHeight": 24,
          "paddingLeft": 5,
          "paddingRight": 5,
          "top": 1.5,
        },
        Object {
          "color": "#424242",
          "fontFamily": "SinkinSans-200XLight",
          "fontSize": 12,
          "paddingBottom": 10,
          "paddingLeft": 5,
          "paddingRight": 10,
          "paddingTop": 10,
        },
      ]
    }
    underlineColorAndroid="rgba(0,0,0,0)"
    virtual={undefined}
  />
</View>
`;

exports[`RENDERINPUT --- Snapshot renders the input with error if touched and error are true 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#E0956D",
      "borderBottomWidth": 1,
      "borderColor": "red",
      "borderLeftWidth": 0,
      "borderRightWidth": 0,
      "borderTopWidth": 0,
      "flexDirection": "row",
      "height": 40,
      "justifyContent": "center",
      "marginLeft": 2,
      "marginTop": "3%",
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
  />
  <TextInput
    allowFontScaling={true}
    editable={true}
    onChangeText={[Function]}
    placeholder="items"
    placeholderTextColor="white"
    secureTextEntry={true}
    style={
      Array [
        Object {
          "color": "#000",
          "flex": 1,
          "fontSize": 17,
          "height": 50,
          "lineHeight": 24,
          "paddingLeft": 5,
          "paddingRight": 5,
          "top": 1.5,
        },
        Object {
          "color": "#424242",
          "fontFamily": "SinkinSans-200XLight",
          "fontSize": 12,
          "paddingBottom": 10,
          "paddingLeft": 5,
          "paddingRight": 10,
          "paddingTop": 10,
        },
      ]
    }
    underlineColorAndroid="rgba(0,0,0,0)"
    virtual={undefined}
  />
</View>
`;

exports[`STYLEDTEXT --- Snapshot renders the styled text with custom font and small font 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  color="red"
  customFont="Sans"
  ellipsizeMode="tail"
  smallFont={true}
  style={
    Array [
      Object {
        "fontFamily": "SinkinSans-200XLight",
        "fontSize": 12,
      },
      Object {
        "color": "red",
        "fontFamily": "Sans",
        "fontSize": 11,
      },
    ]
  }
/>
`;

exports[`STYLEDTEXT --- Snapshot renders the styled text with small font only 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  color="red"
  customFont="Sans"
  ellipsizeMode="tail"
  smallFont={true}
  style={
    Array [
      Object {
        "fontFamily": "SinkinSans-200XLight",
        "fontSize": 12,
      },
      Object {
        "color": "red",
        "fontFamily": "Sans",
        "fontSize": 11,
      },
    ]
  }
/>
`;

exports[`STYLEDTEXT --- Snapshot renders the styled text without custom font only 1`] = `
<Text
  accessible={true}
  allowFontScaling={true}
  color="red"
  customFont="Sans"
  ellipsizeMode="tail"
  style={
    Array [
      Object {
        "fontFamily": "SinkinSans-200XLight",
        "fontSize": 12,
      },
      Object {
        "color": "red",
        "fontFamily": "Sans",
      },
    ]
  }
/>
`;
